python/huggingfaceserver/huggingfaceserver/completions_utils.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2024 The KServe Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+from http import HTTPStatus
+from typing import Dict, List, Optional, Union
+from vllm.engine.async_llm_engine import AsyncLLMEngine
+from kserve.protocol.rest.openai.types.openapi import (
+    CreateCompletionRequest,
+    ErrorResponse,
+    Error,
+    Logprobs,
+)
+from vllm.transformers_utils.tokenizer import get_tokenizer
+from vllm.sequence import Logprob
+
+
+class OpenAIServing:
+
+    def __init__(self, engine: AsyncLLMEngine, served_model: str):
+        self.engine = engine
+        self.served_model = served_model
+
+        self.max_model_len = 0
+        self.tokenizer = None
+
+        try:
+            event_loop = asyncio.get_running_loop()
+        except RuntimeError:
+            event_loop = None
+
+        if (
+            event_loop is not None and event_loop.is_running()
+        ):  # If the current is instanced by Ray Serve, there is already a running event loop
+            event_loop.create_task(self._post_init())
+        else:  # When using single vLLM without engine_use_ray
+            loop = asyncio.new_event_loop()
+            loop.run_until_complete(self._post_init())
+            loop.close()
+
+    async def _post_init(self):
+        engine_model_config = await self.engine.get_model_config()
+        self.max_model_len = engine_model_config.max_model_len
+
+        # A separate tokenizer to map token IDs to strings.
+        self.tokenizer = get_tokenizer(
+            engine_model_config.tokenizer,
+            tokenizer_mode=engine_model_config.tokenizer_mode,
+            trust_remote_code=engine_model_config.trust_remote_code,
```

Comment:
Set the default to false and consider exposing it through a separate PR

Reply:
Yes, there are no changes to accept `trust_remote_code` in this PR. Hence, model_config will default to false. When that change comes in the subsequent PR, it'll get set in the engine args and reflect here.